### PR TITLE
Link to image file should point to original upload regardless of fallback setting

### LINKED ISSUE #2470: Linked images in galleries aren't replaced
## Modern Image Formats Plugin doesn't replace linked images in galleries

I just tested the Modern Image Formats plugin. It appears to work fine for single image blocks, and for galleries where the images aren't linked (link set to: none). In those cases, original images (png) are replaced by AVIF files.

However, when the image link is set to "Link images to media files" or "Enlarge on click", then the AVIF is used inside the `` tag, but the `<a href>` link (to the bigger image format) is still linking to the PNG version.

Those large images would benefit _the most_ from the size reduction of modern formats. Therefore I consider the plugin's feature isn't complete it those images aren't replaced as well.

## Steps to reproduce

1. Create a Post with an Image Gallery, and add some images.
2. Click on the link tool and set "Link images to media files"
2. On the page preview, inspect the source code.

## Additional Context

 - Plugin Version: 2.6.1

### PR DESCRIPTION
**Pull request by faisalahammad**
## Summary

Fixes #2470

## Relevant technical choices

When "Also generate fallback images in the original upload format" is disabled, Modern Image Formats swaps the attachment's main file for the generated modern-format version and backs up the true original in the `original_image` metadata. Because of that, `wp_get_attachment_url()` legitimately returns the modern-format file, and the "Link to image file" href on Image/Gallery blocks (built from the REST `source_url` field) ends up pointing to it instead of the original upload. With fallback enabled this doesn't happen, so the link behavior was inconsistent depending on that one setting - as discussed and agreed on in the issue thread.

This PR filters the REST attachment response so `source_url` (and `media_details.sizes.full.source_url`) point back to the original file when this plugin has swapped formats. Detection compares the mime type of the currently attached file against the backed-up original's mime type, so it only triggers on this plugin's own format swap and leaves core's own unrelated "-scaled" resizing (same mime type) untouched. The `` tag rendering pipeline is untouched too - it still shows the modern format as before, only the link/href changes.

This affects new image insertions going forward, since Gutenberg bakes the href into the block's stored HTML at insert time. Already-published posts keep their existing link until the image is re-inserted or the block is re-saved.

Added `plugins/webp-uploads/tests/test-attachment-url.php` covering: format swap detected, fallback enabled (no-op), core's own scaled images (no-op), and img tag still showing the modern format.

## Use of AI Tools

Used Claude Code as a coding assistant for this PR: investigating the issue thread and the plugin's upload/rewrite pipeline, drafting the fix and its tests. I reviewed the logic, ran the full webp-uploads PHPUnit suite, phpcs, and phpstan locally, and verified the fix and edge cases manually before opening this PR.

### CODE REVIEW

File: plugins/webp-uploads/rest-api.php [hunk 1]
```diff
@@ -79,3 +79,66 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
...
+
+	$attached_file = get_attached_file( $post->ID );
+	if ( false === $attached_file ) {
+		return $response;
+	}
+
+	/*
+	 * Compare the mime type of the currently attached file against the backed-up original, not
+	 * `get_post_mime_type()`: WordPress intentionally keeps the post's mime type as the original
+	 * one for compatibility even after this plugin swaps the attached file. Core's own "-scaled"
```
Comment (adamsilverstein) [inline plugins/webp-uploads/rest-api.php:120]: is that correct? I expect the -scaled size to also be in the output format, not the original upload format.

To test:
- Add a [filter](https://gist.github.com/adamsilverstein/a586a3e2d09dac01495102c199d9a96a) mapping jpeg->webp
- upload an oversized jpeg images
- check the uploads folder for which versions are created

**Comment by adamsilverstein** [comment]
@ms-studio can you check to see if this resolves your issue?

File: plugins/webp-uploads/rest-api.php (hunk 1 above)
Comment (faisalahammad) [reply plugins/webp-uploads/rest-api.php:120]: Good catch. Core's `_wp_image_meta_replace_original()` keeps the `-scaled` suffix on the swapped attached file, so the right target for the link in that case is the scaled output-format file (e.g. `leaves-scaled.webp`), not the unscaled `original_image`.

Updated `webp_uploads_update_rest_attachment_original_source_url()` in `plugins/webp-uploads/rest-api.php`: when the attached file matches `/-scaled(?=\.[^.]+$)/`, the filter now uses `wp_get_attachment_url( $post->ID )` (which resolves to the swapped scaled modern-format file); otherwise it still falls back to `metadata['original_image']` for non-scaled swapped images.

Added `test_source_url_points_to_scaled_output_format_when_fallback_disabled` in `plugins/webp-uploads/tests/test-attachment-url.php` to lock the scaled-fallback-disabled case in. Existing tests for the fallback-enabled and non-scaled cases still pass. phpcs + phpstan + 184 webp-uploads PHPUnit tests all green.

Pushed in commit a324265 on `fix/2470-gallery-link-original-image`.
